nncf/api/compression.py

```diff
@@ -122,16 +122,14 @@ def epoch_step(self, next_epoch: Optional[int] = None) -> None:
             next_epoch = self.current_epoch + 1
         self.current_epoch = next_epoch
 
-    def load_state(self, step: int, epoch: int) -> None:
+    def load_state(self, state: Dict[str, int]) -> None:
         """
         Loads the compression scheduler state, but does not update the state of the
         compression method.
 
-        :param step: The index of the global training step.
-        :param epoch: The training epoch index.
+        :param state: Output of `get_state()` method.
         """
-        self.current_step = step
-        self.current_epoch = epoch
+        self.__dict__.update(state)
```

Comment:
It looks like a potential bug.

Reply:
What is the problem?

Reply:
The explicit assignment is preferable to the implicit assignment. The state value is an object, at least you should make a copy. But I believe the explicit assignment is better.

Reply:
Done